lib/normalize.js

```diff
@@ -70,7 +125,7 @@ const normalize = async (pkg, { strict, steps, root, changes, allowLegacyCase })
         changes?.push(`"bundleDependencies" was changed from an object to an array`)
         data.bundleDependencies = Object.keys(bd)
       }
-    } else {
+    } else if ('bundleDependencies' in data) {
```

Comment:
```suggestion
    } else if (Object.prototype.hasOwnProperty.call(data, 'bundleDependencies')) {
```
to be consistent with other data checks in fixer

Reply:
https://github.com/npm/package-json/blob/main/lib/normalize.js#L304

We do it both ways an `in` is way easier to read imho.